### ISSUE TITLE
Fix Calendar Accessibility with Selected Dates

### ISSUE THREAD

**Issue by amandadupell** [grommet/grommet#5429]
Related to #4068 and #5319 

### Expected Behavior

When the user selects a date or a range of dates, the selected date(s) should be relayed to the user via the screen reader in some way.

### Actual Behavior

When a user selects a date or a range of dates, nothing is relayed to the user that a date was selected or a change occurred.

### Video exhibiting the issue

https://user-images.githubusercontent.com/38971752/124836905-ce372e80-df51-11eb-8342-59026d6c7538.mov

### Steps to Reproduce

1. Use MacBook Pro VoiceOver or another screen reader and open the simple story for Calendar in Storybook
2. Using arrow key commands (control + option + arrow keys), navigate to the table of dates
3. Choose any date by selecting with the space bar or enter key

### Your Environment

- Grommet version: 2.17.3
- Browser Name and version: Chrome
- Operating System and version (desktop or mobile): MacBook Pro Desktop
